### ISSUE TITLE
REQ: `Sketchup::DefinitionList#import_from_url`

### ISSUE THREAD

**Issue by DanRathbun** [SketchUp/api-issue-tracker#900]
#### SketchUp Ruby API Feature Request

The [`Sketchup::DefinitionList`](https://ruby.sketchup.com/Sketchup/DefinitionList.html) class has no `#import_from_url` method compliment to the [`#load_from_url`](https://ruby.sketchup.com/Sketchup/DefinitionList.html#load_from_url-instance_method) method, so that coders can download and import **non-SKP** files from a network or a remote web server using a `LoadHandler` interface.

Attempting to use [`#import`](https://ruby.sketchup.com/Sketchup/DefinitionList.html#import-instance_method) to import from a network raises an `#<IOError: Failed converting file to SKP>` exception.

REF: https://forums.sketchup.com/t/using-definition-import-to-import-dae-files-only-works-locally/230861

----

The current workaround is to use the [`Sketchup::Http::Request`](https://ruby.sketchup.com/Sketchup/Http/Request.html) to download the file to a local disk location and then do the local import using [`Sketchup::DefinitionList#import()`](https://ruby.sketchup.com/Sketchup/DefinitionList.html#import-instance_method). After a successful import, code should probably delete the temporary downloaded file using [`File::delete`](https://ruby-doc.org/core-2.7.2/File.html#method-c-delete).

**Comment by thomthom** [comment]
I think what you describe as a workaround is a reasonable workflow.

**Comment by DanRathbun** [comment]
Yes, of course but it is much more complex for newbs especially if they have no web interaction experience.
